src/test/java/org/jitsi/meet/test/DesktopSharingTest.java

```diff
@@ -37,150 +29,115 @@
     extends WebTestBase
 {
     /**
-     * A property to specified the external script that will be used to
-     * start the new participant.
+     * The xpath for the local video element.
      */
-    public final static String HOOK_SCRIPT = "desktop.sharing.hook.script";
+    private static final String LOCAL_VIDEO_XPATH
+        = "//span[@id='localVideoWrapper']";
 
     /**
-     * Exception on starting the hook, null if script has started successfully.
+     * Check desktop sharing start.
      */
-    private Exception hookException = null;
-
-    @Override
-    public boolean skipTestByDefault()
+    @Test
+    public void testDesktopSharingStart()
     {
-        return true;
+        ensureOneParticipant();
+
+        String extId
+                = (String) getParticipant1().getConfigValue(
+                "desktopSharingChromeExtId");
+
+        if(extId == null) {
+            throw new SkipException(
+                "No Desktop sharing configuration detected. Disabling test.");
+        }
+
+        ensureTwoParticipants(
+                null, null, null,
+                new WebParticipantOptions().setChromeExtensionId(extId));
+        startDesktopSharing();
+        checkExpandingDesktopSharingLargeVideo(true);
+        testDesktopSharingInPresence("desktop");
+
     }
 
     /**
-     * Ensure we have only one participant available in the room - the owner.
-     * Starts the new participants using the external script and check whether
-     * the stream we are receiving from him is screen.
-     * Returns at the end the state we found tests - with 2 participants.
+     * Check desktop sharing stop.
      */
-    @Test
-    public void testDesktopSharingInPresence()
+    @Test(dependsOnMethods = { "testDesktopSharingStart" })
+    public void testDesktopSharingStop()
     {
-        final String hookScript = System.getProperty(HOOK_SCRIPT);
-
-        if (hookScript == null)
-            return;
-
-        print("Start testDesktopSharingInPresence.");
+        stopDesktopSharing();
+        checkExpandingDesktopSharingLargeVideo(false);
+        testDesktopSharingInPresence("camera");
+    }
 
-        ensureOneParticipant();
+    /**
+     * Checks the status of desktop sharing received from the presence and
+     * compares it to the passed expected result.
+     * @param expectedResult camera/desktop
+     */
+    private void testDesktopSharingInPresence(final String  expectedResult)
+    {
+        String participant1Jid
+            = MeetUtils.getResourceJid(getParticipant2().getDriver());
```

Comment:
The variable name suggests this is the id of participant 1, but the code gets the id of participant 2. What is the intention?

Reply:
No intention. It's a mistake! I should change the variable name!

Reply:
I'm merging anyway and will change it